Ukhu/Banka
# #165401163 Implement LFA's feedback on API endpoints

#### What does this PR do?
Implements the feedback gotten from LFA for the backend codebase

#### Description of Task to be completed?
 - restructure tests into separate folders
 - add jsdocs documentation
 - use better naming conventions
 - implement feedback changes to the different routes and controllers
 - use more descriptive error messages

#### How should this be manually tested?
 - Clone the repo
 - run 'npm run dev' to ensure the app is functioning properly
 - run 'npm run test' to ensure all tests are passing
 - cd into the API directory to verify that codebase changes have been implemented

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#165401163

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A
